controller/rancher/rancher.go

```diff
@@ -156,23 +183,46 @@ func (lbc *LoadBalancerController) BuildConfigFromMetadata(lbName, envUUID, self
 		}
 	}
 	frontendsMap := map[string]*config.FrontendService{}
-	// fetch certificates
+
```

Comment:
I'd move this logic to the CertFetcher. There is no need to parse the labels here once CertFetcher is set with the defaultCertDir and CertDir in the Run() method. 

FetchCertificate method would need to change to accept lbMeta *LBMetadata, isDefault boolean flag and return a list of certificates:

```
FetchCertificates(lbMeta *LBMetadata, isDefault boolean) []*config.Certificate
```

Then inside FetchCertificate implementation the decision whether to read from the dir or from the metadata passed, should be made (by checking if defaultDir and certDir) are set. And it should return either non default cert list or list of one default certificate entry.

I just want to minimize certificate fetching logic in Rancher.go provider. Ideally we should just have CertFetcher fields initializing + call to FetchCertificate here.